cnc_ctrl_v1/Kinematics.cpp

```diff
@@ -204,9 +204,21 @@ void  Kinematics::triangularInverse(float xTarget,float yTarget, float* aChainLe
     float Motor1Distance = sqrt(pow((-1*_xCordOfMotor - xTarget),2)+pow((_yCordOfMotor - yTarget),2));
     float Motor2Distance = sqrt(pow((_xCordOfMotor - xTarget),2)+pow((_yCordOfMotor - yTarget),2));
 
-    //Calculate the chain angles from horizontal
-    float Chain1Angle = 3.14159 - acos(R/Motor1Distance) - acos((_yCordOfMotor - yTarget)/Motor1Distance);
-    float Chain2Angle = 3.14159 - acos(R/Motor2Distance) - acos((_yCordOfMotor - yTarget)/Motor2Distance);
+    //Calculate the chain angles from horizontal, based on if the chain connects to the sled from the top or bottom of the sprocket
+    if(sysSettings.chainOverSprocket == 1){
+        float Chain1Angle = asin((_yCordOfMotor - yTarget)/Motor1Distance) + asin(R/Motor1Distance);
```

Comment:
Unfortunately I don't think C++ lets us create the float variable within the `if` statement (python would be OK with it, but C++ is picky)

I think we need to define` float Chain1Angle = 0; float Chain2Angle = 0` just above the `if `

Reply:
Thanks for the catch! I'll fix that now.